### PR TITLE
[5.2] Remove Toolbar::getInstance() from views

### PR DESCRIPTION
**Pull request by Hackwar**
### Summary of Changes
`Toolbar::getInstance()` has been deprecated for quite some time now, but the alternative wasn't really clear yet. This PR replaces all occurences with `$this->getDocument()->getToolbar()` instead. This leaves the `Toolbar::getInstance()` in the `ToolbarHelper` class.

### Testing Instructions
Codereview

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by HLeithner** [comment]
What the issue @brianteeman ?

**Comment by brianteeman** [comment]
> What the issue @brianteeman ?

1. obvious errors due to blindly search/replace (picked up by drone)
2. stating that no documentation is required.

**Comment by HLeithner** [comment]
I checked all files but can't find a search replace issue. if you mean the conversation from `Toolbar::getInstance('toolbar');` to `$this->getDocument()->getToolbar();` without the parameter `toolbar` this is ok because the default toolbar is `toolbar`.

on the side I checked some file for missing removal of the `use` statement for the `toolbar` which can be removed if not used, as it has been done for some view.

**Comment by pe7er** [comment]
@Hackwar Could you please remove the unused ``use Joomla\CMS\Toolbar\Toolbar;`` statements?

**Comment by Hackwar** [comment]
Done

File: administrator/components/com_banners/src/View/Client/HtmlView.php [hunk 1]
```diff
@@ -111,7 +111,7 @@ protected function addToolbar(): void
         $isNew      = ($this->item->id == 0);
         $checkedOut = !(\is_null($this->item->checked_out) || $this->item->checked_out == $user->id);
         $canDo      = $this->canDo;
-        $toolbar    = Toolbar::getInstance();
+        $toolbar    = $this->getDocument()->getToolbar();
```
Comment (Quy) [inline administrator/components/com_banners/src/View/Client/HtmlView.php:114]: Please remove import.

File: administrator/components/com_categories/src/View/Category/HtmlView.php [hunk 2]
```diff
@@ -145,7 +145,7 @@ protected function addToolbar()
 
         $user      = $this->getCurrentUser();
         $userId    = $user->id;
-        $toolbar   = Toolbar::getInstance();
+        $toolbar   = $this->getDocument()->getToolbar();
```
Comment (Quy) [inline administrator/components/com_categories/src/View/Category/HtmlView.php:157]: Please remove import.

File: administrator/components/com_contact/src/View/Contact/HtmlView.php [hunk 3]
```diff
@@ -114,7 +114,7 @@ protected function addToolbar()
         $userId     = $user->id;
         $isNew      = ($this->item->id == 0);
         $checkedOut = !(\is_null($this->item->checked_out) || $this->item->checked_out == $userId);
-        $toolbar    = Toolbar::getInstance();
+        $toolbar    = $this->getDocument()->getToolbar();
```
Comment (Quy) [inline administrator/components/com_contact/src/View/Contact/HtmlView.php:126]: Please remove import.

File: administrator/components/com_content/src/View/Article/HtmlView.php [hunk 4]
```diff
@@ -142,7 +142,7 @@ protected function addToolbar()
         $userId     = $user->id;
         $isNew      = ($this->item->id == 0);
         $checkedOut = !(\is_null($this->item->checked_out) || $this->item->checked_out == $userId);
-        $toolbar    = Toolbar::getInstance();
+        $toolbar    = $this->getDocument()->getToolbar();
```
Comment (Quy) [inline administrator/components/com_content/src/View/Article/HtmlView.php:154]: Please remove import.

**Comment by Quy** [comment]
> Done

I flagged some. There are more to remove.

File: administrator/components/com_banners/src/View/Client/HtmlView.php (hunk 1 above)
Comment (Hackwar) [reply administrator/components/com_banners/src/View/Client/HtmlView.php:114]: The import is used a few lines further down and can not be removed.

File: administrator/components/com_categories/src/View/Category/HtmlView.php (hunk 2 above)
Comment (Hackwar) [reply administrator/components/com_categories/src/View/Category/HtmlView.php:157]: The import is used a few lines further down and can not be removed.

File: administrator/components/com_contact/src/View/Contact/HtmlView.php (hunk 3 above)
Comment (Hackwar) [reply administrator/components/com_contact/src/View/Contact/HtmlView.php:126]: The import is used a few lines further down and can not be removed.

File: administrator/components/com_content/src/View/Article/HtmlView.php (hunk 4 above)
Comment (Hackwar) [reply administrator/components/com_content/src/View/Article/HtmlView.php:154]: The import is used a few lines further down and can not be removed.

**Comment by Hackwar** [comment]
> > Done
> 
> I flagged some. There are more to remove.

No, I thoroughly went through them all at that time and all other instances are still in use.

**Comment by laoneo** [comment]
@Quy the code style checker should also throw an error when there are unused imports. So I guess we are fine here, or?

**Comment by Quy** [comment]
@laoneo Yes. Please discard my comments.

**Comment by laoneo** [comment]
Thanks!

**Pull request merged**